lib/Doctrine/DBAL/Platforms/DateIntervalUnit.php

```diff
@@ -22,7 +26,43 @@
 
     public const YEAR = 'YEAR';
 
-    private function __construct()
+    public static function SECOND() : self
+    {
+        return self::get(self::SECOND);
+    }
+
+    public static function MINUTE() : self
+    {
+        return self::get(self::MINUTE);
+    }
+
+    public static function HOUR() : self
+    {
+        return self::get(self::HOUR);
+    }
+
+    public static function DAY() : self
+    {
+        return self::get(self::DAY);
+    }
+
+    public static function WEEK() : self
+    {
+        return self::get(self::WEEK);
+    }
+
+    public static function MONTH() : self
+    {
+        return self::get(self::MONTH);
+    }
+
+    public static function QUARTER() : self
+    {
+        return self::get(self::QUARTER);
+    }
+
+    public static function YEAR() : self
```

Comment:
why define all these accessors, when the `__callStatic` should take over ?

Reply:
Performance + static analysis.